core/src/test/java/org/apache/iceberg/inmemory/TestInMemoryFileIO.java

```diff
@@ -108,4 +111,18 @@ public void testOverwriteBeforeAndAfterClose() throws IOException {
     inputStream.close();
     Assertions.assertThat(new String(buf)).isEqualTo("new data");
   }
+
+  @Test
+  public void testFilesAreSharedAcrossMultipleInstances() {
+    String location = randomLocation();
+    InMemoryFileIO fileIO = new InMemoryFileIO();
+    fileIO.addFile(location, "hello world".getBytes());
+
+    InMemoryFileIO fileIO2 = new InMemoryFileIO();
+    Assertions.assertThat(fileIO2.fileExists(location)).isTrue();
```

Comment:
Just a nit: I'd put an `as` so that in case the test fails there's a clearer error message.

Assertions.assertThat(fileIO2.fileExists(location)).as("File should still exist in another InMemoryFileIO instance").isTrue()